### DISCUSSION TITLE
Suggestion on how to make a "year in review" with Navidrome data?

### DISCUSSION THREAD

**Discussion by jessienab** [navidrome/navidrome#2664] [Q&A]
I would like to poll my Navidrome server scrobble data from the last year, to then use it to generate my own "Year in Review" (similar to Spotify Wrapped or other such services).

Not entirely sure what the best approach to this is. Going off of "# of Plays" within the webUI isn't that nice because that includes all sorts of music I may not have even played in the last year...

Are scrobbles stored with exact timestamp inside the database? Or is the play count simply incremented on the specific track ID?

I am thinking that if individual scrobbles data is stored, I can poll each track associated with the scrobble, get it's ID Tags and playcount within the last year, then put that all in a CSV esque format for further processing.

Thanks!

**Comment by metalheim** [comment]
## Alternatives:
1. You could scrobble to services like last.fm or musicbrainz and let them handle the "year in Review" aggregation.
2. If you don't like the idea of sharing this data with a 3rd party (understandably), you can absolutely host your own instance of maloja (https://github.com/krateng/maloja) and configure it as custom listenbrainz BaseURL to scrobble to. 

Above solutions dont work retrospectively though. 

## Built-in Statistics / analytics
Haveing listening statistics and analytics built into navidrome would be great, but I don't think navidrome retains detailed playback statistics to support this. And Maloja exists and is pretty much tailormade for this usecase.
If you (or someone else willing) comes up with a solution and opens a pull request, it would be welcome of course :)

**Comment by jessienab** [comment]
> ## Alternatives:
> 
>     1. You could scrobble to services like last.fm or musicbrainz and let them handle the "year in Review" aggregation.
> 
>     2. If you don't like the idea of sharing this data with a 3rd party (understandably), you can absolutely host your own instance of maloja (https://github.com/krateng/maloja) and configure it as custom listenbrainz BaseURL to scrobble to.
>

Thank you for the suggestion for maloja, I think this works perfectly for my use being self-hosted datawise, while still integrating very cleanly in to Navidrome :)

I agree that a built-in statistics page would be nice, but definitely there are more important issues at hand to resolve.

Thank you again for the ideas! :+1:
